cellarium/cas/models.py

```diff
@@ -0,0 +1,188 @@
+            description="The ID of the dataset containing cells", examples=["a7a92fb49-50741b00a-244955d47"]
+        )
+        count_per_dataset: int = Field(description="The number of cells found in the dataset", examples=[10])
+        min_distance: float = Field(
+            description="The minimum distance between the query cell and the dataset cells",
+            examples=[1589.847900390625],
+        )
+        max_distance: float = Field(
+            description="The maximum distance between the query cell and the dataset cells",
+            examples=[1840.047119140625],
+        )
+        median_distance: float = Field(
+            description="The median distance between the query cell and the dataset cells", examples=[1791.372802734375]
+        )
+        mean_distance: float = Field(
+            description="The mean distance between the query cell and the dataset cells", examples=[1791.372802734375]
+        )
+
+    class SummaryStatistics(BaseModel):
+        cell_type: str = Field(description="The cell type of the cluster of cells", examples=["erythrocyte"])
+        cell_count: int = Field(description="The number of cells in the cluster", examples=[94])
+        min_distance: float = Field(
+            description="The minimum distance between the query cell and the cluster cells",
+            examples=[1589.847900390625],
+        )
+        p25_distance: float = Field(
+            description="The 25th percentile distance between the query cell and the cluster cells",
+            examples=[1664.875244140625],
+        )
+        median_distance: float = Field(
+            description="The median distance between the query cell and the cluster cells", examples=[1791.372802734375]
+        )
+        p75_distance: float = Field(
+            description="The 75th percentile distance between the query cell and the cluster cells",
+            examples=[1801.3585205078125],
+        )
+        max_distance: float = Field(
+            description="The maximum distance between the query cell and the cluster cells",
+            examples=[1840.047119140625],
+        )
+        dataset_ids_with_counts: t.Optional[t.List["CellTypeSummaryStatisticsResults.DatasetStatistics"]] = None
+
+    class NeighborhoodAnnotation(BaseModel):
+        """
+        Represents the data object returned by the CAS API for a single nearest neighbor annotation.
+        """
+
+        query_cell_id: str = Field(description="The ID of the querying cell", examples=["ATTACTTATTTAGTT-12311"])
+        matches: t.List["CellTypeSummaryStatisticsResults.SummaryStatistics"]
+
+    data: t.List["CellTypeSummaryStatisticsResults.NeighborhoodAnnotation"] = Field(description="The annotations found")
+
+
+CellTypeSummaryStatisticsResults.model_rebuild()
+
+
+class CellTypeOntologyAwareResults(BaseModel):
+    """
+    Represents the data object returned by the CAS API for a ontology-aware annotations.
+    """
+
+    class Matches(BaseModel):
```

Comment:
Does this class represent multiple matches or just one match?  If it's just one match, could it be `Match`?

Reply:
Gah, you're totally right